src/legacy/ui/public/agg_types/number_list/utils.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { last } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { htmlIdGenerator } from '@elastic/eui';
+
+import { parseRange, Range } from '../../utils/range';
+import { NumberRowModel } from './number_row';
+
+const EMPTY_STRING = '';
+const defaultRange = parseRange('[0,Infinity)');
+const generateId = htmlIdGenerator();
+
+function parse(value: string) {
+  const parsedValue = parseFloat(value);
+  return isNaN(parsedValue) ? EMPTY_STRING : parsedValue;
+}
+
+function getRange(range?: string): Range {
+  try {
+    return range ? parseRange(range) : defaultRange;
+  } catch (e) {
+    throw new TypeError('Unable to parse range: ' + e.message);
+  }
+}
+
+function validateValue(value: number | '', numberRange: Range) {
+  const result = {
+    isValid: true,
+    errors: [] as string[],
+  };
+
+  if (value === EMPTY_STRING) {
+    result.isValid = false;
+  } else if (!numberRange.within(value)) {
+    result.isValid = false;
+    result.errors.push(
+      i18n.translate('common.ui.aggTypes.numberList.invalidRangeErrorMessage', {
+        defaultMessage: 'The value should be in the range: {range}.',
+        values: { range: `[${numberRange.min}, ${numberRange.max}]` },
```

Comment:
Non-technical users won't understand this notation, can we just have it read: `The value should be in the range of 0 to 100`